src/app/link.service.ts

```diff
@@ -23,26 +27,25 @@ export interface LinkData {
 })
 export class LinkService {
 
-  private linkList: LinkData[] = [];
-  dataLoaded = new EventEmitter();
+  linkListSource = new BehaviorSubject<LinkData[]>([]);
+  linkList$ = this.linkListSource.asObservable();
 
-  getLinks(): LinkData[] {
-    return this.linkList;
+  private setLinks(newLinks: LinkData[]): void {
+    this.linkListSource.next(newLinks);
   }
 
-  addLinks(newLinks: LinkData[]): void {
-    for (const str of newLinks) {
-      this.linkList.push(str);
-    }
-  }
+  constructor(private fetchService: FetchStatusService,
+    private ngZone: NgZone) {
 
-  constructor() {
     chrome.tabs.getCurrent((tab) => {
       chrome.tabs.sendMessage(tab.openerTabId, {
         message: 'send links'
       }, (links: LinkData[]) => {
-        this.addLinks(links);
-        this.dataLoaded.emit();
+        this.ngZone.run(() => {
+          this.fetchService.initMap(links);
+          this.setLinks(links);
```

Comment:
My idea was that I wanted to give each link to an observable first initially so that we could display all of the links and then let the status code come in once each one was ready.

Reply:
I guess since everything is passed by reference though it doesn't actually matter